bin/run_controller

```diff
@@ -144,17 +146,14 @@ cid=$(docker run -d --privileged --name $CONTAINER \
 if [ $IMAGE == "forch/controller" ] && [[ $faucet_id =~ ^[0-9]+$ ]]; then
     for file in $INSTDIR/forch/*; do docker cp $file $CONTAINER:/etc/faucet/; done
     # docker exec $CONTAINER /usr/bin/gauge --ryu-config-file=${GAUGE_RYU_CONF} --ryu-ofp-tcp-listen-port=${GAUGE_LISTEN_PORT} --ryu-wsapi-host=${WSAPI_LISTEN_HOST} --ryu-wsapi-port=${WSAPI_LISTEN_PORT} --ryu-app=ryu.app.ofctl_rest &
-    docker exec $CONTAINER /usr/bin/gauge_go &
-    docker exec $CONTAINER /usr/bin/forch &
     pid=$(docker inspect --format="{{ .State.Pid }}" $cid)
     data_intf_name=data_faucet_$faucet_id
     sudo ip link add $data_intf_name type veth peer name data0 netns $pid
     sudo ip link set $data_intf_name up
-    docker exec $CONTAINER ip link add link data0 name data0.272 type vlan id 272
-    docker exec $CONTAINER ip addr add 192.168.2.$faucet_id dev data0.272
+    docker exec $CONTAINER ip addr add 192.168.2.$faucet_id dev data0
     docker exec $CONTAINER ip link set data0 up
-    docker exec $CONTAINER ip link set data0.272 up
-    docker exec $CONTAINER ip route add 192.168.0.0/16 dev data0.272
+    echo Adding fot bridge port
+    docker exec $CONTAINER bash -c "ovs-vsctl add-port br-fot data0 -- set interface data0 ofport_request=1; ifconfig data0 up"
```

Comment:
This part should really go inside the container.  Although it is *triggered* by an external event (data0 appearing), it's not dependent on it and shouldn't be exposed.  One solution would be to add a "while data0 not configured" wait loop (see etc/start_faux for an example) or the docker exec call here would directly execute a "data0_ready" script inside of the container.

The reason is encapsulation. What you've done here is basically exposing container internals externally.

Reply:
Makes sense. Added it to controller_go